### DISCUSSION TITLE
Report error

### DISCUSSION THREAD

**Discussion by kms71271** [netbox-community/netbox#11666] [Q&A] [answered]
I have Netbox v3.4.4 and 'Reports' is not working properly.

```
Server Error
There was a problem with your request. Please contact an administrator.

The complete exception is provided below:

<class 'NameError'>

name 'IPADDRESS_ROLE_LOOPBACK' is not defined

Python version: 3.10.6
NetBox version: 3.4.4
```

Is anyone having same issue?

**Comment by candlerb** [comment, marked as answer]
It looks like an error in your report. Can you show the code?

The string `IPADDRESS_ROLE_LOOPBACK` does not appear anywhere in the current Netbox source code, so I suspect you've taken a report which was written for an older version of Netbox.  I'm afraid Netbox makes no guarantees at all about forward or backward compatibility of any APIs, including the REST API and the reports/scripts APIs. When you upgrade Netbox, it's up to you to follow the release notes and make any necessary changes on your side.

What you probably want is `IPAddressRoleChoices.ROLE_LOOPBACK`.  See example code [here](https://github.com/netbox-community/reports/blob/master/reports/ipam-reports/ip-check-prefix.py).
